lib/wanda/node_instrumentation.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Wanda.Instrumentation do
+  @moduledoc """
+  Remote targets instrumentation module.
+  """
+
+  def trigger_facts_gathering(execution_id, target, checks) do
```

Comment:
This is the tricky part, right?
Because here, we would need to trigger the fact gathering, and wait until the agents sends back the facts.
I see it nicely done with different GenServer calls in any case.
I'm not sure how the DynamicServer plays here though hehe

Reply:
The `DynamicSupervisor` allows us to spawn an indefinite number of executions, where a normal `Supervisor` accepts static children.

Whatever we call the method, here is where I imagine Wanda publishing towards the agents messages about starting facts gathering for the checks.

Here's a touch-point with the Agent and your comment about the
```
# sbd_config is a hardcoded key to identify which fact gathering to do
# keys is a list with the keys from the list to return
[{"name": "sbd_config", "keys": ["SBD_DEVICE", "SBD_TIMEOUT_ACTION"]}]
```
What information do we have at hand?
What are we gonna send to the Agent?
Most likely here we need to make some steps ahead with the DSL itself

Reply:
Then, for the receiving (waiting) part, I imagine a component listening for incoming facts gathered and calls the apppropriate method on the ExecutionServer (what we simulate now)
```elixir
Wanda.ExecutionServer.collect_gathered_facts(...)
```
(Gonna improve naming)